api.go

```diff
@@ -492,6 +509,10 @@ func (c Client) dumpHTTP(req *http.Request, resp *http.Response) error {
 
 // do - execute http request.
 func (c Client) do(req *http.Request) (*http.Response, error) {
+	c.counter.Incr(1)
+	if c.counter.Rate() > RequestRateLoggingTh {
+		c.log.Warnf("minio requests-per-second execceded %d", RequestRateLoggingTh)
```

Comment:
This will 1) not give us the exact number 2) would potentially flood logs.
You need to start a goroutine in `NewWithOptions` with a timer that will check this counter once a second and if it's over the limit, it will write how much.